### PR TITLE
Enhance Itanium demangler interface.

### PR DESCRIPTION
**Pull request by kbobrovs**
Add new APIs needed to use the demangler for C++ demangling in IR passes.
This is a prerequisite for the upcoming ESIMD LLVM passes.

Signed-off-by: Konstantin S Bobrovsky <konstantin.s.bobrovsky@intel.com>

### CODE REVIEW

File: llvm/include/llvm/Demangle/ItaniumDemangle.h [hunk 1]
```diff
@@ -2085,6 +2089,13 @@ class IntegerLiteral : public Node {
     if (Type.size() <= 3)
       S += Type;
   }
+
+  // Retrievs the string view of the integer value represented by this node.
```
Comment (bader) [inline llvm/include/llvm/Demangle/ItaniumDemangle.h:2093]:
```suggestion
  // Retrieves the string view of the integer value represented by this node.
```

File: llvm/include/llvm/Demangle/ItaniumDemangle.h [hunk 2]
```diff
@@ -2085,6 +2089,13 @@ class IntegerLiteral : public Node {
...
+  const StringView &getValue() const { return Value; }
+
+  // Retrievs the string view of the type string of the integer value this node
+  // represents.
```
Comment (bader) [inline llvm/include/llvm/Demangle/ItaniumDemangle.h:2097]:
```suggestion
  // Retrieves the string view of the type string of the integer value this node
  // represents.
```

**Comment by kbobrovs** [comment]
OK, will do. Reduced the patch to the minimum required.

**Comment by kbobrovs** [comment]
Phabricator:
https://reviews.llvm.org/D81905

**Comment by bader** [comment]
Thanks.
Let's hope it will be accepted w/o regression tests and usages inside the project code base.

Please, fix `buildbot/sycl-win-x64-pr` failure.

**Comment by kbobrovs** [comment]
> Please, fix buildbot/sycl-win-x64-pr failure.

The failure is not related to this patch. Just rebuilt, and this time it went fine.

**Pull request merged**